internal/persistence/queries.go

```diff
@@ -492,51 +490,126 @@ LIMIT ?;
 	return tLE, nil
 }
 
-func DeleteEntry(db *sql.DB, entry *types.TaskLogEntry) error {
-	secsSpent := entry.SecsSpent
-
-	tx, err := db.Begin()
-	if err != nil {
-		return err
-	}
-	defer func() {
-		_ = tx.Rollback()
-	}()
-
-	stmt, err := tx.Prepare(`
+func DeleteTaskLogEntry(db *sql.DB, entry *types.TaskLogEntry) error {
+	return runInTx(db, func(tx *sql.Tx) error {
+		stmt, err := tx.Prepare(`
 DELETE from task_log
 WHERE ID=?;
 `)
-	if err != nil {
-		return err
-	}
-	defer stmt.Close()
+		if err != nil {
+			return err
+		}
+		defer stmt.Close()
 
-	_, err = stmt.Exec(entry.ID)
-	if err != nil {
-		return err
-	}
+		_, err = stmt.Exec(entry.ID)
+		if err != nil {
+			return err
+		}
 
-	tStmt, err := tx.Prepare(`
+		tStmt, err := tx.Prepare(`
 UPDATE task
 SET secs_spent = secs_spent-?,
     updated_at = ?
 WHERE id = ?;
     `)
-	if err != nil {
+		if err != nil {
+			return err
+		}
+		defer tStmt.Close()
+
+		_, err = tStmt.Exec(entry.SecsSpent, time.Now().UTC(), entry.TaskID)
 		return err
+	})
+}
+
+func runInTxAndReturnID(db *sql.DB, fn func(tx *sql.Tx) (int64, error)) (int64, error) {
+	tx, err := db.Begin()
+	if err != nil {
+		return -1, err
+	}
+
+	lastID, err := fn(tx)
+	if err == nil {
+		return lastID, tx.Commit()
+	}
+
+	rollbackErr := tx.Rollback()
+	if rollbackErr != nil {
+		return lastID, fmt.Errorf("%w: %w: %s", ErrCouldntRollBackTx, rollbackErr, err.Error())
 	}
-	defer tStmt.Close()
 
-	_, err = tStmt.Exec(secsSpent, time.Now().UTC(), entry.TaskID)
+	return lastID, err
+}
+
+func runInTx(db *sql.DB, fn func(tx *sql.Tx) error) error {
+	tx, err := db.Begin()
 	if err != nil {
 		return err
 	}
 
-	err = tx.Commit()
+	err = fn(tx)
+	if err == nil {
+		return tx.Commit()
+	}
+
+	rollbackErr := tx.Rollback()
+	if rollbackErr != nil {
+		return fmt.Errorf("%w: %w: %s", ErrCouldntRollBackTx, rollbackErr, err.Error())
```

Comment:
```suggestion
		return fmt.Errorf("%w: %w: %w", ErrCouldntRollBackTx, rollbackErr, err)
```

Or simply this


```suggestion
		return errors.Join(ErrCouldntRollBackTx, rollbackErr, err)
```

Otherwise you won't be able to catch an error such as duplicate key or something in such a case.